nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -1186,6 +1223,50 @@ contract StakingEscrow is Issuer, IERC900History {
         emit CommitmentMade(staker, nextPeriod, lockedTokens);
     }
 
+    /**
+    * @notice Migrate from the old period length to the new one. Can be done only once
+    * @param _staker Staker
+    */
+    function migrate(address _staker) public {
+        StakerInfo storage info = stakerInfo[_staker];
+        // check that provided address is/was a staker
+        require(info.subStakes.length != 0 || info.lastCommittedPeriod != 0);
+        if (info.flags.bitSet(MIGRATED_INDEX)) {
+            return;
+        }
+
+        // reset state
+        info.currentCommittedPeriod = 0;
+        info.nextCommittedPeriod = 0;
+        // maintain case when no more sub-stakes and need to avoid re-registering this staker during deposit
+        info.lastCommittedPeriod = 1;
+        info.workerStartPeriod = recalculatePeriod(info.workerStartPeriod);
+        delete info.pastDowntime;
+
+        // recalculate all sub-stakes
+        for (uint256 i = 0; i < info.subStakes.length; i++) {
+            SubStakeInfo storage subStake = info.subStakes[i];
+            subStake.firstPeriod = recalculatePeriod(subStake.firstPeriod);
+            // sub-stake has fixed last period
+            if (subStake.lastPeriod != 0) {
+                subStake.lastPeriod = recalculatePeriod(subStake.lastPeriod);
+                subStake.unlockingDuration = 0;
+            // sub-stake has no fixed ending but possible that with new period length will have
+            } else {
+                uint16 oldCurrentPeriod = uint16(block.timestamp / genesisSecondsPerPeriod);
+                uint16 lastPeriod = recalculatePeriod(oldCurrentPeriod + subStake.unlockingDuration);
+                subStake.unlockingDuration = lastPeriod - getCurrentPeriod();
+                if (subStake.unlockingDuration == 0) {
+                    subStake.lastPeriod = lastPeriod;
+                }
+            }
+        }
+
+        policyManager.migrate(_staker);
+        info.flags = info.flags.toggleBit(MIGRATED_INDEX);
+        emit Migrated(_staker);
```

Comment:
So migrations will happen during the first commitment made by a worker after the contract has been upgraded - does it make sense to include the period in the `Migrated` event, since this can be varied? I guess the question would be is the period number value based on the old calculation or new calculation based on hours per period... - presumably the same period value we use for the `CommitmentMade` event generated by this commitment?

Reply:
We could, just not sure. If for global statistics - then makes sense

Reply:
Interested to hear what others think - but seems beneficial in my mind.

Reply:
✔️